sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaCoder.java

```diff
@@ -140,7 +140,14 @@ private void verifyDeterministic(Schema schema)
 
   @Override
   public boolean consistentWithEquals() {
-    return true;
+    // We can't guarantee that the user type T has a good equals method, so we assume the coder is
+    // not consistent with equals (BEAM-8364).
+    return false;
+  }
```

Comment:
We've generally said elsewhere that user schema types must have equalsTo defined. Should we just document that requirement?

Reply:
My only concern with that is if a user's type doesn't have equals defined they'll just get undefined behavior, and not a useful error message. Could we reasonably detect when a user type doesn't meet that requirement and throw an error (or log a warning and have this function return false)?

I think it's technically possible, my first draft of this PR looked at the typeDescriptor and only returned true when the class had equals defined, but I was concerned it was brittle since we don't know the user-defined equals is actually a good one.